README.md

```diff
@@ -48,3 +48,68 @@ Any use of third-party trademarks or logos are subject to those third-party's po
 ## libazureinit
 
 For common library used by this reference implementation, please refer to [libazureinit](https://github.com/Azure/azure-init/tree/main/libazureinit/).
+
+## Tracing Logic Overview
+
+### How Tracing is Set Up
+
+The tracing setup in this project is built around three key layers, each with its own responsibility:
+
+1. **EmitKVPLayer**: Custom Layer for Span Processing
+2. **OpenTelemetryLayer**: Context Propagation and Span Export
+3. **stderr_layer**: Formatting and Logging to stderr
+
+These layers work together, yet independently, to process span data as it flows through the program.
+
+### Layer Overview
+
+#### 1. EmitKVPLayer
+
+- **Purpose**: This custom layer is responsible for processing spans by capturing their metadata, generating key-value pairs, encoding them into a specific format, and writing the encoded data to the VM's hyperV file. 
```

Comment:
Maybe drop a link to 

https://learn.microsoft.com/en-us/virtualization/hyper-v-on-windows/reference/integration-services#hyper-v-data-exchange-service-kvp

Indicate this is meant for consumption by `hv_kvp_daemon` service, which typically reads from `/var/lib/hyperv/.kvp_pool_1`.